src/api/api-hooks.ts

```diff
@@ -19,6 +19,7 @@ export const useCartData = () => {
 export const useCartTotals = () => {
     const ecomApi = useEcomAPI();
     const { data } = useCartData();
+    const [isUpdatingOnCartChange, setIsUpdatingOnCartChange] = useState(false);
```

Comment:
do we really care if it's loading only due to cart change? can't we use `cartTotals.isLoading`?

Reply:
`cartTotals.isLoading` - this is the period when data is fetching for the first time
`cartTotals.isValidating` - this is the period when data is fetching to update the cache and it works also on 
_SWRConfig_
```
   revalidateOnFocus: true,
   revalidateOnReconnect: true,
   refreshInterval: 5 * 60_000, // 5 minutes
```

Using `isValidating` makes checkout button blinking (from normal to disable state) on each onFocus and it look strange.

Reply:
we fetch cart totals each time user hovers on the `Checkout` button?

Reply:
no, browser tab onFocus

Reply:
The option I see is to remove that `useEffect` that mutates cart totals on cart change.
If cart totals depend on cart, then we should reflect in key https://swr.vercel.app/docs/conditional-fetching
For example, using cart updated date:
```tsx
const { data } = useCartData();
const cartTotals = useSwr('cart-totals' + data._updatedDate', fn)

cartTotals.isLoading; // true when cart totals refetch after cart change, not on revalidate
```

Reply:
@Letushev let me test it

Reply:
@Letushev I've played with it a bit.
`() =>` `'cart-totals/${data._updatedDate}` - this generates many caches. We will need to clean old cache. 
`() => cart ? 'cart-totals' : null` - this doesn't react on cart change.

Reply:
Okay, your solution or `isValidating` is fine then

Reply:
> maybe it's ok to have the button disabled in such case

Agree. Kept `isValidating`